### PR TITLE
Removed missing verse warnings for Verse Spans

### PR DESCRIPTION
**Pull request by mannycolon**
#### Describe what your pull request addresses (Please include relevant issue numbers):
-

#### Please include detailed Test instructions for your pull request:
- [ ] Test with builds in: https://github.com/unfoldingWord/translationCore/actions/runs/1388420994 
- [ ] Import a project with a verse span such as (https://git.door43.org/lrsallee/en_ust_gal_book, https://git.door43.org/christopherrsmith/en_ust_act_book, https://git.door43.org/lrsallee/en_ust_luk_book, https://git.door43.org/lrsallee/en_ust_jos_book)
- [ ] The missing verses warning should not appear.
- [ ] Now open a project with missing verses but no verse spans (https://git.door43.org/manny_colon/en_xxx_jos_book)
- [ ] The missing verses warning should appear for Joshua 14:3
and Joshua 14:4.
- [ ] Now open a project with both missing verses and verse spans (https://git.door43.org/manny_colon/en_ust_act_book)
- [ ] The missing verses warning should appear for Acts 1:23 and not for Acts 1:24 and Acts 1:25 (The verse span).

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword/translationcore/7179)

### CODE REVIEW

**Comment by PhotoNomad0** [comment]
@mannycolon Seeing a problem with GEN (https://git.door43.org/MattCarlton/en_ust_gen_book):
Screen Shot 2021-10-27 at 12 39 14 PM

**Comment by PhotoNomad0** [comment]
It looks like it doesn't handle verse spans of more than 2 verses.

**Comment by PhotoNomad0** [comment]
Forgot to tell you there is a library function you can call to get the verse range: 
const { low, hi } = verseHelpers.getVerseRangeFromSpan(verse_);
import { verseHelpers } from 'tc-ui-toolkit';

File: src/js/helpers/ProjectValidation/MissingVersesHelpers.js [hunk 1]
```diff
@@ -28,11 +28,39 @@ export const getMissingVerses = (projectDir, bookAbbr, expectedVerses) => {
...
+              let results = verseText.match(pattern);
+              const result = results ? results[0] : null;
+              let isBetween = false;
+
+              if (result) {
+                // Verses between starting and ending numbers of verse span
+                const numbers = result.split('-');
+                const number1 = parseInt(numbers[0]);
+                const number2 = parseInt(numbers[1]);
+                isBetween = number1 < verseIndex && number2 > verseIndex;
```
Comment (PhotoNomad0) [inline src/js/helpers/ProjectValidation/MissingVersesHelpers.js]: Why not just test `number1 <= verseIndex && number2 >= verseIndex` and simplify the logic below?

**Comment by PhotoNomad0** [comment]
@mannycolon 
~Still seeing warning when I open projects I previously imported.  Will start over loading apps in released version of app (3.0.1) and then upgrading to this build.~

Everything worked well when I tried the upgrade scenario above.

File: src/js/helpers/ProjectValidation/MissingVersesHelpers.js (hunk 1 above)
Comment (mannycolon) [reply src/js/helpers/ProjectValidation/MissingVersesHelpers.js]: Great catch, I actually missed that. Thanks.

**Review by PhotoNomad0 — APPROVED** [review]

**Pull request merged**